web/src/inbox_ui.ts

```diff
@@ -668,9 +675,69 @@ export function complete_rerender(): void {
         return;
     }
     load_data_from_ls();
+
+    const narrow_stream_name = current_filter()?.operands("channel")[0];
+    if (narrow_stream_name) {
+        $("#inbox-pane").html(render_inbox_view({
+            normal_view: false,
+            search_val: search_keyword,
+            INBOX_SEARCH_ID,
+        }));
+        const stream_id = stream_data.get_stream_id(narrow_stream_name);
+        // TODO: Handle case where stream_id is undefined.
+        assert(stream_id !== undefined);
+        const widget = new TopicListWidget($("#inbox-list"), stream_id);
+        const formatter = (conversation: TopicInfo): ListInfoNode => {
+            const render = (): string => render_inbox_row({
+                is_stream: false,
+                conversation_key: get_topic_key(stream_id, conversation.topic_name),
+                is_collapsed: false,
+                is_direct: false,
+                is_hidden: false,
+                is_topic: true,
+                topic_url: conversation.url,
+                topic_name: conversation.topic_name,
+                mention_in_unread: conversation.contains_unread_mention,
+                stream_id,
+                unread_count: conversation.unread,
+                all_visibility_policies: user_topics.all_visibility_policies,
+                visibility_policy: user_topics.get_topic_visibility_policy(stream_id, conversation.topic_name),
+            });
+
+            const eq = (other: ListInfoNode): boolean =>
+                other.type === "topic" && _.isEqual(conversation, other.conversation);
+
+            const key = "t:" + conversation.topic_name;
+
+            return {
+                key,
+                render,
+                type: "topic",
+                conversation,
+                eq,
+            };
+        }
+        widget.build(true, formatter);
+        stream_topic_history_util.get_server_history(stream_id, () => {
+            widget.build(false, formatter);
+        });
```

Comment:
@timabbott looking for you thoughts on this before proceeding further. What do you think about using exisitng `TopicListWidget` for this? I am not sure doing this from scrach would be that much cleaner either?

Reply:
I think this is probably a fine direction; what you should do if you're doing this is probably to do a `TopicListWidget` base class with `LeftSidebarTopicListWidget` and `CenterPaneTopicListWidget` subclasses.